src/elastio-snap.c

```diff
@@ -3420,13 +3420,35 @@ static int snap_trace_bio(struct snap_device *dev, struct bio *bio){
 	struct tracing_params *tp = NULL;
 	sector_t start_sect, end_sect;
 	unsigned int bytes, pages;
+	int max_sectors;
 
 	//if we don't need to cow or physical memory usage has exceeded threshold or
 	//	COW file state is failed, this bio just call the real mrf normally
 	if (!bio_needs_cow(bio, dev) || memory_is_too_low(dev) || tracer_read_cow_fail_state(dev)) {
 		return elastio_snap_call_mrf(dev->sd_orig_mrf, bio);
 	}
 
+	max_sectors = queue_max_sectors(bdev_get_queue(dev->sd_base_dev));
+	if (bio_sectors(bio) > max_sectors) {
+#ifdef HAVE_BIO_SPLIT_4
+		struct bio *split = bio_split(bio, max_sectors,
+				GFP_NOIO, dev_bioset(dev));
+		bio_chain(split, bio);
+#ifdef HAVE_GENERIC_MAKE_REQUEST
+		generic_make_request(bio);
+#else
+		submit_bio_noacct(bio);
+#endif
```

Comment:
Why don't we just do `bio_put(bio);` for the original bio?

Reply:
Because we still need the other part of the original bio to be processed. Hence, we cannot just dismiss it